### PR TITLE
CAP-346 extract metadata

### PR DESCRIPTION
**Pull request by tomw66**
Here's what I've done so far for https://metoffice.atlassian.net/browse/CAP-346, if you want to take a look and check I'm on the right track!

### CODE REVIEW

File: tests/unit/data/test_extract_metadata.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+		height = DimCoord(3.5, 
+					standard_name="height",
+					units="m",
+					attributes={'positive': 'up'})
+		cube = Cube(np.zeros((4, 8, 24), np.float32),
+					standard_name="mass_concentration_of_ozone_in_air",
+					units="ug/m3",
+					dim_coords_and_dims=[(latitude, 0),
+										(longitude, 1),
+										(time, 2)])
```
Comment (crbunney) [inline tests/unit/data/test_extract_metadata.py:30]: One minor thing, the formatting in this file is a bit awkward. In Pycharm I'd just use the autoformatter, apparently VSCode has some options: https://code.visualstudio.com/docs/python/editing#_formatting

Has this file been run through a formatter?

**Review by crbunney — COMMENTED** [review]
Looks good. It all gets pretty complex pretty fast considering all the possible variations of iris cubes and cubelists, so I'm thinking that once you're happy with it, we should get this in and start using it to snag any issues based on actual use.

File: environment.yml [hunk 2]
```diff
@@ -22,6 +22,8 @@ dependencies:
   - flake8
   - boto3-stubs-essential
   - pip:
+    - git+ssh://git@github.com/MetOffice/edr_server@CAP-340_add_collection_deserialisation
```
Comment (crbunney) [inline environment.yml:25]: This repo is public now, so you can switch the URL to the `https` version and it should fix the issue with the build not running properly due to permissions

File: tests/unit/data/test_extract_metadata.py (hunk 1 above)
Comment (tomw66) [reply tests/unit/data/test_extract_metadata.py:30]: Won't let me install any :( 
image
Autoformatter works fine for javascript. Remind me is there a different location/method to install it?

File: environment.yml [hunk 3]
```diff
@@ -22,6 +22,8 @@ dependencies:
   - flake8
   - boto3-stubs-essential
   - pip:
+    - git+https://git@github.com/MetOffice/edr_server@CAP-340_add_collection_deserialisation
```
Comment (tomw66) [inline environment.yml:25]: @crbunney Now getting 'ModuleNotFoundError: No module named 'edr_server'', is the URL still wrong?

File: tests/unit/data/test_extract_metadata.py (hunk 1 above)
Comment (crbunney) [reply tests/unit/data/test_extract_metadata.py:30]: Looks like it's trying to install into the system-wide location, which you won't have permissions to do. Try using `/bin/python3 -m pip install --user -U autopep8` (The `--user` flag will install it just for you in your user area)

File: environment.yml (hunk 3 above)
Comment (crbunney) [reply environment.yml:25]: oh yea, try this format: `https://github.com/{user}/{repository}/archive/{branch}.zip` 

e.g. https://github.com/MetOffice/edr_server/archive/CAP-340_add_collection_deserialisation.zip

**Comment by crbunney** [comment]
I've been investigating why the build is failing.

the `edr_server` install appears to be working fine based on running these commands:
```bash
conda env remove --name tmp

conda create --name tmp python=3.8 pip ipython -y && conda activate tmp

conda install -y proj==8.0.0
pip install https://github.com/MetOffice/edr_server/archive/CAP-340_add_collection_deserialisation.zip
ipython -c 'import edr_server' # No output if successful, otherwise expect a traceback

pip install https://github.com/ADAQ-AQI/clean-air-project/archive/CAP-346-extract-metadata.zip


ipython -c 'from clean_air.data.data_filter_handler import DataFilterHandler'  # No output if successful
```

I suspect the issue is to do with how tox is creating the environment, as tox isn't using the `environment.yml`.

I suspect we'll have to tell `tox` where to get the dependency from, by doing something similar to [this](https://github.com/sqlobject/sqlobject/blob/ee26d9787b276686f5be9a19229134f3256e24a3/tox.ini#L23) in the `tox.ini`

File: tox.ini [hunk 4]
```diff
@@ -4,5 +4,7 @@ isolated_build = True
 
 [testenv]
 extras = dev
-deps = pytest
+deps = 
+	pytest
+	edr_server: git+https://github.com/MetOffice/edr_server.git@CAP-340_add_collection_deserialisation#egg=edr_server
```
Comment (tomw66) [inline tox.ini:9]: @crbunney Hasn't seemed to have had any affect, does my implementation seem ok?

**Pull request merged**